src/cache/cache_las.c

```diff
@@ -1186,72 +1194,102 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
+         * Expect an update entry with:
+         *  1. Not in a prepare locked state
+         *  2. Durable timestamp as not max timestamp
+         *  3. For an in-progress prepared update, durable timestamp should be zero
+         *  and no restriction on durable timestamp value for other updates.
          */
-        if (las_btree_id != saved_btree_id || saved_key->size != las_key.size ||
-          memcmp(saved_key->data, las_key.data, las_key.size) != 0) {
-            /* If we've examined enough entries, give up. */
-            if (cnt == 0)
-                break;
+        WT_ERR_ASSERT(session, prepare_state != WT_PREPARE_LOCKED, EINVAL,
+          "LAS prepared record is in locked state");
+        WT_ERR_ASSERT(session, durable_timestamp != WT_TS_MAX, EINVAL,
+          "LAS record durable timestamp is set as MAX timestamp");
+        WT_ERR_ASSERT(
+          session,
+          ((prepare_state != WT_PREPARE_INPROGRESS || durable_timestamp == 0) &&
+            (prepare_state == WT_PREPARE_INPROGRESS || durable_timestamp >= las_timestamp)),
+          EINVAL,
+          "Either LAS record is a in-progress prepared update with wrong durable timestamp or not"
+          " a prepared update with wrong durable timestamp. prepared state: %" PRIu8
+          " durable timestamp: %" PRIu64,
+          prepare_state, durable_timestamp);
 
-            saved_btree_id = las_btree_id;
-            WT_ERR(__wt_buf_set(session, saved_key, las_key.data, las_key.size));
+        /*
+         * Check to see if the page or key has changed this iteration, and if they have, setup
+         * context for safely removing obsolete updates.
+         */
+        remove_entry = false;
+        if (prev_entry_can_remove) {
+            if (las_btree_id != saved_btree_id || saved_key->size != las_key.size ||
+              memcmp(saved_key->data, las_key.data, las_key.size) != 0) {
+                remove_entry = true;
+                saved_btree_id = las_btree_id;
+                WT_ERR(__wt_buf_set(session, saved_key, las_key.data, las_key.size));
+            } else if (upd_type != WT_UPDATE_MODIFY)
+                remove_entry = true;
+            else
+                skip_update_modify_cnt++;
+        }
 
-            /*
-             * Expect an update entry with:
-             *	1. not in a prepare locked state
-             *	2. durable timestamp as not max timestamp.
-             *	3. for an in-progress prepared update, durable
-             *	timestamp should be zero.
-             *	4. no restriction on durable timestamp value
-             *	for other updates.
-             */
-            WT_ASSERT(session, prepare_state != WT_PREPARE_LOCKED &&
-                durable_timestamp != WT_TS_MAX &&
-                (prepare_state != WT_PREPARE_INPROGRESS || durable_timestamp == 0));
+        /* Remove the obsolete entries */
+        if (remove_entry) {
+            /* Remove each individual LAS record till the current record. */
```

Comment:
This comment no longer makes sense, "till the current record" isn't what we do now.

Reply:
> That was what I was thinking (add the reset only when __las_sweep_init() called`), I'm OK with your preference if you disagree.

Done.

> This comment no longer makes sense, "till the current record" isn't what we do now.

Removed.